src/dmd/lexer.d

```diff
@@ -201,8 +202,22 @@ class Lexer
         return peek(t).value;
     }
 
+    /****************************
+     * Turn next token in buffer into a token, including newlines.
+     * Params:
+     *  t = the token to set the resulting Token to
+     */
+    final void scanUntilNewline(Token* t)
+    {
+        this.tokenizeNewlines++;
+        scan(t);
+        this.tokenizeNewlines--;
```

Comment:
I don't see a reason this needs to be nested. tokenizeNewlines just needs to be set when a `#` is encountered, and reset when an end of line token is seen. That should be all that is necessary.